src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -233,6 +233,8 @@ public function findOrNew($id, $columns = ['*'])
      */
     public function firstOrNew(array $attributes)
     {
+        $attributes = $this->model->newInstance()->fill($attributes)->attributesToArray();
```

Comment:
yes if someone tries to use variables that aren't fillable, but if that's the case it should not create a new model anyway because those attributes are guarded